src/components/fileDropArea/hooks/useFileProcessing.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2025 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ChangeEvent, useCallback, useState } from 'react';
+
+import {
+  FileValidationError as ValidationError,
+  FileValidationMessages,
+  FileValidationOptions,
+  FileWithValidation,
+} from '../types';
+import { validateFile } from '../utils';
+
+interface FileHandlingOptions extends FileValidationOptions {
+  messages: FileValidationMessages;
+  onFilesAdded: (files: FileWithValidation[]) => void;
+}
+
+export const useFileHandling = ({
+  maxFileSize = 134217728,
+  acceptFileMimeTypes = [],
+  onFilesAdded,
+}: FileHandlingOptions) => {
+  const [errorMessage, setErrorMessage] = useState<ValidationError | null>(null);
+
+  const processFiles = useCallback(
+    (files: File[]) => {
+      setErrorMessage(null);
+
+      const validatedFiles: FileWithValidation[] = files.map((file) => ({
+        file,
+        validationErrors: validateFile(file, { maxFileSize, acceptFileMimeTypes }),
+      }));
+
+      onFilesAdded(validatedFiles);
+
+      const firstErrorResult = validatedFiles.find(
+        ({ validationErrors }) => validationErrors.length,
+      );
+
+      if (firstErrorResult) {
+        setErrorMessage(firstErrorResult.validationErrors[0]);
+      }
+    },
+    [maxFileSize, acceptFileMimeTypes, onFilesAdded],
+  );
+
+  const onDrop = useCallback(
+    (acceptedFiles: File[]) => {
+      processFiles(acceptedFiles);
+    },
+    [processFiles],
+  );
+
+  const handleFileInputChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const { files } = event.target;
+
+      if (files) {
+        processFiles([...files]);
+      }
+
+      const target = event.target as HTMLInputElement;
+
+      if (target) {
+        target.value = '';
+      }
```

Comment:
```suggestion
    target?.value = '';
```

Reply:
TS2779: The left-hand side of an assignment expression may not be an optional property access.